apps/site/src/pages/settings/billing/upgrade/change-payment-method-modal.tsx

```diff
@@ -0,0 +1,287 @@
+      open={open}
+      onClose={onClose}
+      contentStyle={{
+        top: "40%",
+      }}
+      data-testid="change-payment-method-modal"
+    >
+      <Box>
+        <Typography variant="bpHeading4" sx={{ marginBottom: 3 }}>
+          Change payment method
+        </Typography>
+        <Typography variant="bpBodyCopy" sx={{ textTransform: "uppercase" }}>
+          <strong>Payment Cards</strong>
+        </Typography>
+        {paymentMethods?.map((paymentMethod) => {
+          const isDefault = currentDefaultPaymentMethodId === paymentMethod.id;
+
+          const isChosen = updatedDefaultPaymentMethodId
+            ? updatedDefaultPaymentMethodId === paymentMethod.id
+            : isDefault;
+
+          return (
+            <Button
+              key={paymentMethod.id}
+              variant="transparent"
+              onClick={() =>
+                setUpdatedDefaultPaymentMethodId(
+                  isDefault ? undefined : paymentMethod.id,
+                )
+              }
+              sx={({ palette, spacing, transitions }) => ({
+                display: "block",
+                textAlign: "start",
+                width: "100%",
+                padding: spacing(0, 1),
+                "& svg.card-logo-icon": {
+                  transition: transitions.create("color"),
+                },
+                ":hover": {
+                  background: palette.purple[10],
+                  "& svg.card-logo-icon": {
+                    color: palette.purple[50],
+                  },
+                  "& p.card-brand": {
+                    color: palette.purple[70],
+                  },
+                  "& span.switch-text": {
+                    opacity: 1,
+                  },
+                },
+              })}
+            >
+              <PaymentMethod
+                paymentMethod={paymentMethod}
+                sx={{ marginY: 1 }}
+                titleEndAdornment={
+                  <Box component="span" position="relative">
+                    <Fade in={isChosen}>
+                      <Box
+                        component="span"
+                        sx={({ palette, spacing }) => ({
+                          padding: spacing(0.5, 0.75),
+                          color: palette.purple[700],
+                          background: palette.purple[20],
+                          borderRadius: 1,
+                          textTransform: "uppercase",
+                          fontSize: 11,
+                        })}
+                      >
+                        <FontAwesomeIcon
+                          icon={faCheck}
+                          sx={{ marginRight: 0.75, fontSize: 12 }}
+                        />
+                        <strong>Chosen</strong>
+                      </Box>
+                    </Fade>
+                    {isChosen ? null : (
+                      <Box
+                        component="span"
+                        className="switch-text"
```

Comment:
Also this one

Reply:
https://github.com/blockprotocol/blockprotocol/blob/8dc5c992ae5b78b6d98a2d900200afaaf1a8de5e/apps/site/src/pages/settings/billing/upgrade/change-payment-method-modal.tsx#L156-L158